dali/python/nvidia/dali/data_node.py

```diff
@@ -122,6 +142,73 @@ def __bool__(self):
                 " represent computations in DALI Pipeline see the \"Mathematical Expressions\""
                 " section of DALI documentation."))
 
+    def __getitem__(self, val):
+        idxs = []
+        new_axes = []
+        new_axis_names = []
+
+        # returns True if this index adds a new output dimension
+        def process_index(idx, dim):
+            if idx is None:
+                print("DUPA!")
+                idxs.append((None, None, None, None))
+                return True
+            elif isinstance(idx, slice):
+                if idx.step is not None and idx.step != 1:
+                    raise NotImplementedError("Slicing with non-unit step is not implemented.")
+                idxs.append((None, idx.start, idx.stop, None))
+                return True
+            elif isinstance(idx, _NewAxis):
+                new_axes.append(dim)
+                if idx.name is not None:
+                    new_axis_names.append(idx.name)
+                return True
+            elif idx is Ellipsis:
+                raise NotImplementedError("Ellipsis in indexing is not implemented")
+            elif isinstance(idx, (float, str)):
+                raise TypeError("Invalid type for an index: ", type)
+            else:
+                idxs.append((idx, None, None, None))
+                return False
+
+        if not isinstance(val, tuple):
+            process_index(val, 0)
+        else:
+            d = 0
+            for v in val:
+                if process_index(v, d):
+                    d += 1
+
+        if len(new_axis_names) != 0:
+            if len(new_axis_names) != len(new_axes):
+                raise ValueError("New axis name must be specified for all axes or none.");
+            new_axis_names = "".join(new_axis_names)
+        else:
+            new_axis_names = None
+
+        slice_args = {}
+        for i, (at, lo, hi, step) in enumerate(idxs):
+            if at   is not None: slice_args["at_%i"%i] = at
+            if lo   is not None: slice_args["lo_%i"%i] = lo
+            if hi   is not None: slice_args["hi_%i"%i] = hi
+            if step is not None: slice_args["step_%i"%i] = step
+
+        import nvidia.dali.fn
+        if len(slice_args) == 0:
+            if len(new_axes) == 0 or new_axes[-1] < len(idxs):
+                print("Adding dim check for ", len(idxs))
```

Comment:
Do we need these prints?

Reply:
Surely not. These are remnants from initial testing - I wanted to be 100% sure that the code follows the intended path and that the right checks fire when expected.